### PR TITLE
updated postResults endpoint to fetch the user_id so that it works wi…

### PR DESCRIPTION
**Pull request by ErikBoonprakong**
…th only the username as an argument

### CODE REVIEW

File: chat.js [hunk 1]
```diff
@@ -13,16 +13,28 @@ const io = require("socket.io")(server, {
 });
 const port = 4000;
 const cors = require("cors");
-
+const clients = [];
```
Comment (megwat) [inline chat.js:16]: could this cause issues in future rooms? possibly add a feature for multiple rooms?

File: chat.js [hunk 2]
```diff
@@ -13,16 +13,28 @@ const io = require("socket.io")(server, {
...
   });
+  socket.on("join room", (name) => {
+    if (!clients.includes(name)) {
+      clients.push(name);
+    }
+    io.emit("player list", clients);
+    if (clients.length === 2) {
+      io.emit("new message", `${clients[0]} vs ${clients[1]}`);
+    }
+  });
```
Comment (megwat) [inline chat.js:33]: this is good but is the best way to deal with this?

File: serverLocal.js [hunk 3]
```diff
@@ -258,8 +258,14 @@ async function getSavedGamesById(server) {
...
 async function postResult(server) {
-  const { user_id, username, won, lost, draw, score } = await server.body;
+  const { username, won, lost, draw } = await server.body;
   let finalScore = await calculateScore(won, lost, draw);
+  let user_id = [
+    ...db
+      .query(`SELECT id FROM users WHERE username = ?`, [username])
+      .asObjects(),
+  ][0].id;
+  console.log(user_id);
```
Comment (megwat) [inline serverLocal.js:268]: can you use {user_id} = ... so you dont have to use [0]

File: serverLocal.js [hunk 4]
```diff
@@ -281,7 +287,11 @@ async function calculateScore(win, lost, draw) {
 
 async function getScores(server) {
   const scores = [
-    ...db.query(`SELECT * FROM leaderboard ORDER BY score DESC`).asObjects(),
+    ...db
+      .query(
+        `SELECT username, SUM(won) as won, SUM(lost) as lost, SUM(draw) as draw, SUM(SCORE) as score FROM leaderboard  GROUP BY user_id ORDER BY score DESC`
+      )
```
Comment (megwat) [inline serverLocal.js:293]: nice sql query

**Review by megwat — APPROVED** [review]

File: serverLocal.js (hunk 4 above)
Comment (ErikBoonprakong) [reply serverLocal.js:293]: thanks

**Pull request merged**